src/DeviceFileSystem.ts

```diff
@@ -0,0 +1,63 @@
+import { BaseFileSystem, BFSCallback, FileSystem, FileSystemOptions
+       } from "browserfs/dist/node/core/file_system";
+import Stats from 'browserfs/dist/node/core/node_fs_stats';
+import { File } from "browserfs/dist/node/core/file";
+import { FileFlag } from "browserfs/dist/node/core/file_flag";
+import { ApiError } from 'browserfs/dist/node/core/api_error';
+
+export interface Device {
```

Comment:
So Device is a completely new concept, it is not in BrowserFS. I see that it has a 'File'

I think we will have to implement a 'File' interface, lets say 'JsaddleFile'.
So why have 'Device', I think doing 'files: {[name: string]: File};' would be sufficient?

Reply:
something like this https://github.com/jvilk/BrowserFS/blob/28a26cbcb89b5cef59a586f55045aabe98561f47/src/backend/Emscripten.ts#L34
https://github.com/jvilk/BrowserFS/blob/28a26cbcb89b5cef59a586f55045aabe98561f47/src/backend/Emscripten.ts#L253